ssh/tailssh/accept_env_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package tailssh
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestMatchPattern(t *testing.T) {
+	testCases := []struct {
+		pattern string
+		target  string
+		match   bool
+	}{
+		// Positive cases
+		{pattern: "*", target: "EXAMPLE_ENV", match: true},
+		{pattern: "?", target: "A", match: true},
+		{pattern: "?*", target: "EXAMPLE_2", match: true},
+		{pattern: "??", target: "CC", match: true},
+		{pattern: "?*?", target: "ABCDEFG", match: true},
+		{pattern: "*?*", target: "ABCDEFG", match: true},
+		{pattern: "***", target: "123456", match: true},
+		{pattern: "**?TEST", target: "_TEST", match: true},
+		{pattern: "TEST**?", target: "TEST_", match: true},
+		{pattern: "TEST_*", target: "TEST_A", match: true},
+		{pattern: "EXAMPLE_?_ENV", target: "EXAMPLE_A_ENV", match: true},
+		{pattern: "EXAMPLE_*_ENV", target: "EXAMPLE_aBcd2231---_ENV", match: true},
+		{pattern: "COMPLICA?ED_PATTERN*", target: "COMPLICATED_PATTERN_REST", match: true},
+		{pattern: "COMPLICAT???ED_PATT??ERN", target: "COMPLICAT123ED_PATTggERN", match: true},
+		{pattern: "DIRECT_MATCH", target: "DIRECT_MATCH", match: true},
+
+		// Negative cases
+		{pattern: "?", target: "123", match: false},
+		{pattern: "??", target: "123", match: false},
+		{pattern: "?*?", target: "A", match: false},
+		{pattern: "*?*", target: "C", match: false},
```

Comment:
Consider including these two tests

```
{pattern: "*", target: "", match: false},
{pattern: "?*", target: "A", match: false},
```

Reply:
I would expect the below to pass, but it fails

```
{pattern: "*?", target: "A", match: false},
```

Reply:
Discussed offline, the current behaviour mirrors openSSH's logic for handling patterns so will stay as-is.

Namely:
- `?` matches one character
- `*` matches zero, one, or many

Reply:
Additional test cases will be brought over from https://github.com/openssh/openssh-portable/blob/46d1fb16b20e971b9ac15e86a3d3e350b49c9ad6/regress/unittests/match/tests.c (openSSH's test suite for their implementation) to catch any remaining edge cases / to ensure compatibility.